src/components/atoms/social-auth-card.tsx

```diff
@@ -0,0 +1,91 @@
+import {
+  Button,
+  Icon,
+  Paper,
+  Stack,
+  SxProps,
+  Theme,
+  Typography,
+} from '@mui/material';
+import useTranslation from 'next-translate/useTranslation';
+import { useLocalStorage } from 'react-use';
+import { useEffect, useState } from 'react';
+import { CheckedButton } from './check-button';
+import { brandColors } from '@/theme';
+
+type Props = {
+  icon: JSX.Element;
+  title: string;
+  description: string;
+  connectHandler?: AccountHandlerConnection;
+  sx?: SxProps<Theme>;
+};
+
+export type AccountHandlerConnection = {
+  isConnected: boolean;
+  connect: any;
+  disconnect: any;
+  isLoading: boolean;
+};
+
+export function SocialAuthCard({
+  icon,
+  title,
+  description,
+  connectHandler,
+  sx,
+}: Props) {
+  const { t } = useTranslation('settings');
+  const [connectedState, setConnectedState] = useState(false);
+
+  useEffect(() => {
+    setConnectedState(connectHandler?.isConnected);
+  }, [connectHandler?.isConnected]);
```

Comment:
try to remove this useEffect. if you're starting the state based on the initial connected state, try set the initial value on the state

```suggestion
  const [connectedState, setConnectedState] = useState(!!connectHandler?.isConnected);
```

if not, comment a follow up pls